src/main/java/org/jabref/gui/mergeentries/newmergedialog/FieldRowController.java

```diff
@@ -0,0 +1,139 @@
+
+    private final String rightValue;
+
+    private final ToggleGroup toggleGroup = new ToggleGroup();
+
+    public FieldRowController(String fieldName, String leftValue, String rightValue, int rowIndex) {
+        fieldNameCell = new FieldNameCell(fieldName, rowIndex);
+        leftValueCell = new FieldValueCell(leftValue, rowIndex);
+        rightValueCell = new FieldValueCell(rightValue, rowIndex);
+        mergedValueCell = new MergedFieldCell(StringUtil.isNullOrEmpty(leftValue) ? rightValue : leftValue, rowIndex);
+
+        this.leftValue = leftValue;
+        this.rightValue = rightValue;
+
+        toggleGroup.getToggles().addAll(leftValueCell, rightValueCell);
+        toggleGroup.selectToggle(StringUtil.isNullOrEmpty(leftValue) ? rightValueCell : leftValueCell);
+        toggleGroup.selectedToggleProperty().addListener(invalidated -> {
+            if (toggleGroup.getSelectedToggle() != null) {
+                mergedValueCell.setText((String) toggleGroup.getSelectedToggle().getUserData());
+            }
+        });
+
+        mergedValueCell.textProperty().addListener((observable, old, mergedValue) -> {
+            if (mergedValue.equals(leftValue)) {
+                toggleGroup.selectToggle(leftValueCell);
+            } else if (mergedValue.equals(rightValue)) {
+                toggleGroup.selectToggle(rightValueCell);
+            } else {
+                // deselect all toggles because left and right values don't equal the merged value
+                toggleGroup.selectToggle(null);
+            }
+        });
+
+        //  When both the left and right cells have the same value, only the left value is displayed,
+        //  making it unnecessary to keep allocating memory for the right cell.
+        if (hasEqualLeftAndRightValues()) {
+            // Setting this to null so the GC release the memory allocated to the right cell.
+            this.rightValueCell = null;
+        }
+    }
+
+    public void selectLeftValue() {
+        toggleGroup.selectToggle(leftValueCell);
+    }
+
+    public void selectRightValue() {
+        if (isRightValueCellHidden()) {
+            selectLeftValue();
+        } else {
+            toggleGroup.selectToggle(rightValueCell);
+        }
+    }
+
+    public String getMergedValue() {
+        return mergedValueProperty().getValue();
+    }
+
+    public ReadOnlyStringProperty mergedValueProperty() {
+        return mergedValueCell.textProperty();
+    }
+
+    public FieldNameCell getFieldNameCell() {
+        return fieldNameCell;
+    }
+
+    public FieldValueCell getLeftValueCell() {
+        return leftValueCell;
+    }
+
+    public FieldValueCell getRightValueCell() {
+        return rightValueCell;
+    }
+
+    public MergedFieldCell getMergedValueCell() {
+        return mergedValueCell;
+    }
+
+    public boolean hasEqualLeftAndRightValues() {
+        return isRightValueCellHidden() || (!StringUtil.isNullOrEmpty(leftValueCell.getText()) &&
```

Comment:
This should be moved to a separate viewModel

Reply:
It is moved in pr B